src/Spectre.Data/Datasets/RoiDataset.cs

```diff
@@ -0,0 +1,65 @@
+﻿/*
+ * Roi.cs
+ * Class represeting dataset with data of regions of interest.
+
+   Copyright 2017 Roman Lisak
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Spectre.Data.Datasets
+{
+    /// <summary>
+    /// Dataset containing information about a single roi.
+    /// </summary>
+    public class RoiDataset
+    {
+        /// <summary>
+        /// Gets or sets the name.
+        /// </summary>
+        /// <value>
+        /// The name.
+        /// </value>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the roi pixels.
+        /// </summary>
+        /// <value>
+        /// The roi pixels.
+        /// </value>
+        public List<RoiPixel> RoiPixels { get; set; }
```

Comment:
What happens if we create `RoiDataset `with list containing pixels whose coordinates exceed the `Width` or `Height` which are stored separately? If we want to have everything here read-only, such thing could be checked in the constructor.